src/libraries/Microsoft.PowerFx.Core/Types/DefinedTypeDependencyGraph.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Parser;
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Syntax.Visitors;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Types
+{
+    internal class DefinedTypeDependencyGraph
+    {
+        private readonly IEnumerable<DefinedType> _definedTypes;
+        private readonly ReadOnlySymbolTable _globalTypes;
+
+        // stores unresolvedType -> its dependencies
```

Comment:
>stores [](http://example.com/codeflow?start=11&length=6)

very minor Nit: for most comments in this pr, they should start with a capital letter